src/freertos/CMakeLists.txt

```diff
@@ -1,3 +1,14 @@
 if(NOT COMPILE_ESP)
-    add_subdirectory(stub)
+    add_subdirectory(posix)
+else()
+    add_library(swarmus_freertos_esp INTERFACE)
+
+    #Hack to match the same include path as posix
+    get_target_property(FREERTOS_SOURCE_DIR idf::freertos SOURCE_DIR)
+    target_include_directories(swarmus_freertos_esp
+            INTERFACE
+            ${FREERTOS_SOURCE_DIR}/include/freertos
+            )
+    target_link_libraries(swarmus_freertos_esp INTERFACE idf::freertos)
+    add_library(SwarmUs::HiveConnect::FreeRTOS ALIAS swarmus_freertos_esp)
```

Comment:
Nice that you got it working, just put it in a subfolder like posix, even if it's just a CMakeLists.txt. 

You could probably invert your if too (thus removing the NOT), so it's easier to add a new platform.

Reply:
Done